### PR TITLE
chore(sdk): Refactor and move all v2 related code to under the v2 namespace.

### PR DESCRIPTION
**Pull request by neuromage**
Most of the changes are around imports and restructuring of the codebase. While it looks like a lot of code was added, most of the code already existed and was simply moved or copied over to v2. The only exceptions are:
- under kfp/v2/components/component_factory.py: some helper functions were copied with simplification from _python_op.py
- we no longer strip the `_path` suffix in v2 components.

Note: there is still some duplication of code (particularly between component_factory.py and _python_op.py), but it's ok for now since we intend to replace some of this with v2 ComponentSpec + BaseComponent.

### CODE REVIEW

File: sdk/python/kfp/compiler/v2_compatible_compiler_test.py [hunk 1]
```diff
@@ -13,142 +13,143 @@
...
+@component
 def preprocess(uri: str, some_int: int, output_parameter_one: OutputPath(int),
                output_dataset_one: OutputPath('Dataset')):
-  '''Dummy Preprocess Step.'''
-  with open(output_dataset_one, 'w') as f:
-    f.write('Output dataset')
-  with open(output_parameter_one, 'w') as f:
-    f.write("{}".format(1234))
+    '''Dummy Preprocess Step.'''
+    with open(output_dataset_one, 'w') as f:
```
Comment (chensun) [inline sdk/python/kfp/compiler/v2_compatible_compiler_test.py:33]: nit: should we keep the 2-space indentation?

File: sdk/python/kfp/v2/components/types/artifact_types.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+
+_GCS_LOCAL_MOUNT_PREFIX = '/gcs/'
+_MINIO_LOCAL_MOUNT_PREFIX = '/minio/'
+_S3_LOCAL_MOUNT_PREFIX = '/s3/'
+
+
+class Artifact(object):
+    """Generic Artifact class.
+
+  This class is meant to represent the metadata around an input or output
```
Comment (chensun) [inline sdk/python/kfp/v2/components/types/artifact_types.py:30]: nit: inconsistent indentation. Should we preserve 2-space indentation? or do you think we want to switch to 4-space indentation going forward?

File: sdk/python/kfp/compiler/v2_compatible_compiler_test.py (hunk 1 above)
Comment (neuromage) [reply sdk/python/kfp/compiler/v2_compatible_compiler_test.py:33]: Let's switch to 4-space, to be consistent with external Google style guide. WDYT?

File: sdk/python/kfp/v2/components/types/artifact_types.py (hunk 2 above)
Comment (neuromage) [reply sdk/python/kfp/v2/components/types/artifact_types.py:30]: Yes, I'm thinking we should use 4-space going forward for consistency with external Google styles and PEP. WDYT?

Comment (chensun) [reply sdk/python/kfp/v2/components/types/artifact_types.py:30]: SGTM

**Pull request merged**